solc/CommandLineParser.cpp

```diff
@@ -997,21 +997,11 @@ void CommandLineParser::processArgs()
 		return;
 
 	checkMutuallyExclusive({g_strColor, g_strNoColor});
+	checkMutuallyExclusive({g_strStopAfter, g_strGas});
 
-	array<string, 9> const conflictingWithStopAfter{
-		CompilerOutputs::componentName(&CompilerOutputs::binary),
-		CompilerOutputs::componentName(&CompilerOutputs::ir),
-		CompilerOutputs::componentName(&CompilerOutputs::irAstJson),
-		CompilerOutputs::componentName(&CompilerOutputs::irOptimized),
-		CompilerOutputs::componentName(&CompilerOutputs::irOptimizedAstJson),
-		g_strGas,
-		CompilerOutputs::componentName(&CompilerOutputs::asm_),
-		CompilerOutputs::componentName(&CompilerOutputs::asmJson),
-		CompilerOutputs::componentName(&CompilerOutputs::opcodes),
-	};
-
-	for (auto& option: conflictingWithStopAfter)
-		checkMutuallyExclusive({g_strStopAfter, option});
+	for (string const& option: CompilerOutputs::componentMap() | ranges::views::keys)
+		if (option != CompilerOutputs::componentName(&CompilerOutputs::astCompactJson))
```

Comment:
Ha, could've added `ranges::views::filter` for `CompilerOutputs::componentName(&CompilerOutputs::astCompactJson)`, but the syntax is atrocious, so it's more readable this way I think.

I had to dig through the code a bit to see what these helper functions do, but from what I can see - `astCompactJson` is the only output that doesn't need to go further than parsing, right? If that's the case, this is a much nicer implementation.

Reply:
That's actually exactly what I tried to do at first :) Those filters ended up being too complex for my taste so I then scrapped them for a simpler approach since AST is the only output allowed in this mode.